packages/eas-cli/src/commands/secrets/delete.ts

```diff
@@ -30,30 +36,51 @@ Unsure where to find the secret's ID? Run ${chalk.bold('eas secrets:list')}`;
     const projectDir = (await findProjectRootAsync()) ?? process.cwd();
     const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
     const projectId = await getProjectIdAsync(exp);
+    const projectFullName = await getProjectFullNameAsync(exp);
+    const projectAccountName = await getProjectAccountNameAsync(exp);
 
     if (!(await isEasEnabledForProjectAsync(projectId))) {
       warnEasUnavailable();
       process.exitCode = 1;
       return;
     }
 
-    const {
+    let {
       args: { id },
     } = this.parse(EnvironmentSecretDelete);
+    let name;
+
+    if (!id) {
+      const secrets = await EnvironmentSecretsQuery.all(projectAccountName, projectFullName);
+
+      const { secret } = await promptAsync({
+        type: 'autocomplete',
+        name: 'secret',
+        message: 'Pick the secret to be deleted:',
+        choices: secrets.map(secret => ({
+          title: `${secret.name} (${secret.scope})`,
+          value: secret,
+        })),
+      });
+
+      ({ id, name } = secret);
+    }
 
     Log.addNewLineIfNone();
     Log.warn(
-      `You are about to permamently delete secret with id: "${id}".\nThis action is irreversible.`
+      `You are about to permamently delete secret${
+        name ? ` "${name}"` : ''
+      } with id: "${id}".\nThis action is irreversible.`
     );
```

Comment:
Maybe check if `secret.scope === ACCOUNT` and then print a warning saying that it'll affect other project builds too.